man/timesyncd.conf.xml

```diff
@@ -103,6 +103,17 @@
         Defaults to 30 seconds and must not be smaller than 1 second.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>SaveIntervalSec=</varname></term>
+        <listitem><para>Interval to periodically save current time. This is especially useful for offline
+        systems with no RTC battery, as it will guarantee that the time increases monotonically between
+        reboots.</para>
+
+        <para>Takes a time span value. The default unit is seconds, but other units may be specified, see
+        <citerefentry><refentrytitle>systemd.time</refentrytitle><manvolnum>5</manvolnum></citerefentry>.
+        Defaults to 0 seconds, meaning disable this option.</para></listitem>
```

Comment:
Shouldn't this be done by default? i.e. pick something like 30s or so?

Reply:
I'd prefer if setting this to "infinity" would turn this off, rather than setting this to "0". I mean if you want something to never happen "infinity" is kinda the right interval. 0 otoh would mean "do this all the time", but that's not intended here...

The parsers already deal with this, and understand "infinity" and map that to USEC_INFINITY

Reply:
> Shouldn't this be done by default? i.e. pick something like 30s or so?

Didn't want to change the default behavior, so opted for this to be implicitly disabled. IMHO most people have RTC clocks and access to NTP servers and won't care for the extra disk writes this option will force on them.

OK to keep at "infinity"?

Reply:
hmm, then maybe let's make this smart: let's chekc if there's an RTC, and if so, set this to infinity, otherwise to 30s or so.

i.e. check if `dir_is_empty("/sys/class/rtc/")` holds. If so we have no RTC.

Reply:
> hmm, then maybe let's make this smart: let's check if there's an RTC, and if so, set this to infinity, otherwise to 30s or so.

Added as a separate, but see my comment above.

Reply:
> i.e. check if dir_is_empty("/sys/class/rtc/") holds. If so we have no RTC.

FYI I think that condition might not be quite as successful as envisioned. Many of the systems that face the issues we are trying to address here have an RTC included in the SoC, it simply has no battery backing present on the board to allow it to persist and increment time when the system is off.

I had the feeling that almost all systems in this category are like this, but I'd have to double check before making such a bold statement, it does look like Raspberry Pi is different to such expectation (it has no RTC at all, unless you add one externally).

Reply:
hmm, but what's the point of an RTC chip if it has no battery? It's a pretty broken concept, no? It's just dead silicon, no? Isn't the right fix for that to simply not load the drivers for that? What does software have as benefit if it loads a driver for hw that misses the one piece of hw it actually needs to be useful?

Reply:
> but what's the point of an RTC chip if it has no battery

It might sound silly, but I have seen a use case of having an accurate external RTC (eg, [DS3231](https://datasheets.maximintegrated.com/en/ds/DS3231.pdf)) with no battery, just to make sure time doesn't drift.

Reply:
Over the past few years I had to deal with [ODROID-N2](https://www.hardkernel.com/blog-2/odroid-n2/) and [LattePanda](https://www.lattepanda.com/). Both come with on-board RTC, but you have to BYOB.

Reply:
> hmm, but what's the point of an RTC chip if it has no battery? 

Use cases include:
- Programmable timed wakeup source from suspend (RTC is always powered during suspend, even when there is no battery, as far as I have seen)
- Some battery-less boards will keep the RTC ticking when the system is off, provided that external power is still connected

Reply:
Yeah got one of those too, many datacenters are grumpy about having battiers around, so it's not that uncommon. Completely losing power is rare enough anyway, most of the times power is retained so the RTC keeps ticking.